std/datetime.d

```diff
@@ -1075,6 +1075,66 @@ public:
         //assert(ist.opCmp(ist) == 0);
     }
 
+    /**
+     * Returns: A hash of the $(LREF SysTime)
+     */
+    size_t toHash() const @nogc pure nothrow @safe
+    {
+        static if (is(size_t == ulong))
+        {
+            return _stdTime;
+        }
+        else
+        {
+            // MurmurHash2
+            enum ulong m = 0xc6a4a7935bd1e995UL;
+            enum ulong n = m * 16;
+            enum uint r = 47;
+
+            ulong k = _stdTime;
+            k *= m;
+            k ^= k >> r;
+            k *= m;
+
+            ulong h = n;
+            h ^= k;
+            h *= m;
+
+            return cast(size_t) h;
+        }
+    }
+
+    @safe unittest
+    {
+        assert(SysTime(0).toHash == SysTime(0).toHash);
+        assert(SysTime(0, LocalTime()).toHash == SysTime(0, LocalTime()).toHash);
+        assert(SysTime(0, LocalTime()).toHash == SysTime(0, UTC()).toHash);
+
+        assert(
+            SysTime(
+                DateTime(2000, 1, 1), LocalTime()
+            ).toHash == SysTime(
+                DateTime(2000, 1, 1), LocalTime()
+            ).toHash
+        );
+
+        immutable zone = new SimpleTimeZone(dur!"minutes"(60));
+        assert(
+            SysTime(
+                DateTime(2000, 1, 1, 1), zone
+            ).toHash == SysTime(
+                DateTime(2000, 1, 1), UTC()
+            ).toHash
+        );
+        assert(
+            SysTime(
+                DateTime(2000, 1, 1), LocalTime()
+            ).toHash != SysTime(
+                DateTime(2000, 1, 1), UTC()
+            ).toHash
```

Comment:
This test will fail for some folks in Europe. In general, tests that rely on `LocalTime` are error prone, and if it's testing that it doesn't match another time zone, then it will fail in some parts of the globe.

Regardless, I think that you're missing my point. You've supposedly implemented MurmurHash2, and yet you have done pretty much nothing to test that you implemented it correctly. Also, all of your tests relate to the time zone and the fact that it's not supposed to be taken into account. None of them relate to different times having different hashes. And while different times will obviously result result in the same hash in some cases (at least on 32-bit systems), you can test at least a few to verify that they're not all spitting out the same value, and it should be easy to avoid the ones that are supposed to have the same value given that hash conflicts should be fairly rare (particularly since you know what hash function you're dealing with).